src/controllers/SlideController.ts

```diff
@@ -1,10 +1,46 @@
 import { NextFunction, Response } from 'express'
+
 import { CreateSlideRequest } from '../interfaces/SlideInterface'
 import SlideService from '../services/SlideService'
-import { slideSchema } from '../validation/slideValidation'
+import { slideSchema, shortenLinkSchema } from '../validation/slideValidation'
 import { validator } from '../validation/validator'
 import GenericError from '../helpers/GenericError'
-import { DeleteSlideReqeust } from '../interfaces/SlideInterface'
+import { DeleteSlideRequest } from '../interfaces/SlideInterface'
+import { FileUploadRequest } from '../interfaces/SlideInterface'
+import SlideHelpers from '../helpers/SlideHelpers'
+
+const uploadSlide = async (
+  req: FileUploadRequest,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const file = req.file
+    const { id } = res.locals.user
+
+    const secureLink = await SlideService.uploadSlide(file)
+
+    const shortenLink = await SlideHelpers.shortenLink(secureLink)
+
+    const validateShortLink = await validator({
+      schema: shortenLinkSchema,
+      data: { shortenLink }
+    })
+
+    if (!validateShortLink.isValid) {
+      throw new GenericError(validateShortLink.error)
+    }
+
+    await SlideService.addSlideToUser(id, shortenLink)
+
+    res.status(200).send({ message: 'success' })
+  } catch (error: unknown) {
+    const exception = error as Error
+
+    if (exception.name !== 'GenericError') return next(exception)
+    res.status(400).json({ message: exception.message })
+  }
+}
 
 const createSlide = async (
```

Comment:
well no we have two controllers missed up what is the difference between uploadSlide & createSlide? :thinking: 

I think it should squashed into one!